To DO:
- A test should be add to accertain `/eos/` or `/afs/` can be accessed before trying the following test: `test_afs.py `, `test_eos.py`, `test_fs.py`, `test_fs_api.py`.
- There is a recursion error from `xaux/fs/fs.py` and `xaux/fs/eos.py` when trying to test from `eos/`:
```Python
_test_helpers.py:50: in init_file
    with ProtectFile(fname, "w", wait=1) as pf:
../xaux/protectfile.py:194: in __init__
    arg['file'] = FsPath(arg['file']).resolve()
../xaux/fs/fs.py:42: in __new__
    return EosPath.__new__(EosPath, *args, _eos_checked=True)
../xaux/fs/eos.py:113: in __new__
    self._set_eos_path()
../xaux/fs/eos.py:123: in _set_eos_path
    parts = _non_strict_resolve(self.expanduser().parent, _as_posix=True).split('/')
/afs/cern.ch/work/t/thpugnat/private/python/miniforge/envs/mf-xtest/lib/python3.10/pathlib.py:873: in parent
    return self._from_parsed_parts(drv, root, parts[:-1])
../xaux/fs/fs.py:101: in _from_parsed_parts
    return FsPath(super(Path, cls)._from_parsed_parts(*args, **kwargs))
../xaux/fs/fs.py:42: in __new__
    return EosPath.__new__(EosPath, *args, _eos_checked=True)
../xaux/fs/eos.py:113: in __new__
    self._set_eos_path()
../xaux/fs/eos.py:123: in _set_eos_path
    parts = _non_strict_resolve(self.expanduser().parent, _as_posix=True).split('/')
/afs/cern.ch/work/t/thpugnat/private/python/miniforge/envs/mf-xtest/lib/python3.10/pathlib.py:873: in parent
    return self._from_parsed_parts(drv, root, parts[:-1])
../xaux/fs/fs.py:101: in _from_parsed_parts
    return FsPath(super(Path, cls)._from_parsed_parts(*args, **kwargs))
E   RecursionError: maximum recursion depth exceeded while calling a Python object
!!! Recursion detected (same locals & position)
```
- Another error is also raised at the same time (see bellow). This is explain by the fact that the error occurs before `_file` was assign in ` ProtectFile`. This will raise an error in the pop section of `release()`.

--- a/xaux/protectfile.py
+++ b/xaux/protectfile.py
@@ -565,7 +565,7 @@ class ProtectFile:
                 self._print_debug("release", f"unlink {self.lockfile}")
                 self.lockfile.unlink()
         # Remove file from the protected register
-        if pop:
+        if pop and hasattr(self, '_file'):
             protected_open.pop(self._file, 0)
 
 

